### DISCUSSION TITLE
load component specfic js file

### DISCUSSION THREAD

**Discussion by qunshen** [oqtane/oqtane.framework#1873] [Q&A]
I have a component which needs a js file and only the component need it.I do not want to put this js file in module or theme. how can I accomplish this?

**Comment by hishamco** [comment]
Could you please check the resources APIs may be it allow the component to participate in resources rendering

**Reply by qunshen** [reply]
could you provide more info? or provide sample code? thanks

**Comment by sbwalker** [comment]
@qunshen what is the purpose of the component and why is it not part of a module or theme? The only example I can think of in the framework is the RichTextEditor - however in the cases where it is used within a module, its static resources must be declared in the module ( Oqtane.Client\Modules\HtmlText\Edit.razor ).

**Reply by qunshen** [reply]
the component is video streaming control which will has a lot java script. and the control will be used in diferent modules!

**Reply by qunshen** [reply]
I found out a solution: create a dedicated theme that has the required resources declared.  any page which uses the video control will be set using the theme! I will try thsi approch. The purpose of my question is that I do not want the java script overhead where video streaming not used.

**Comment by qunshen** [comment]
another idea.  add a new feature which allows developer set additional resources at page level, is that feasible? if we can set resources at page level, my problem can be easily solved. I can see that develoer may be using more special third part components that have special javascript which are only used in that component.

**Reply by sbwalker** [reply]
@qunshen I am not sure you understand the resources concept, as it already supports what you are looking for. 

The default behavior when you declare resources in a module it means that they they will only be included in the specific page where the module is used. If you navigate to a different page where the module is not in use, the resources are removed. 

Note that there is a property in the resource object which allows you to indicate that a resource is global and should be included on all pages - but this is not the default behavior.

**Reply by qunshen** [reply]
thank you for the reply. I will organize virdeo-streaming intto different modules....
